version: '3.7'

services:
  db:
    image: registry.nextgis.com/postgis:latest
    expose:
      - 5432
    volumes:
      - db_dir:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=QMS_TEST_DB
      - POSTGRES_USER=QMS_TEST_DB_USER
      - POSTGRES_PASSWORD=QMS_TEST_DB_USER_PASS


  qqms:
    depends_on:
      - db
    build:
      context: ../
      dockerfile: ./docker/Dockerfile
    image: registry.nextgis.com/qms_dj22py38:1.0.6
    environment:
      - DATABASE_NAME=QMS_TEST_DB
      - DATABASE_USER=QMS_TEST_DB_USER
      - DATABASE_PASS=QMS_TEST_DB_USER_PASS
      - DATABASE_HOST=db
      - DATABASE_PORT=5432
      - EMAIL_HOST=prod.vpn.nextgis.net
      - EMAIL_PORT=587
      - EMAIL_HOST_USER=service@nextgis.com
      - EMAIL_HOST_PASSWORD=sbOqj96DwIX7avIe
      - DEFAULT_FROM_EMAIL=qms@nextgis.com
      - EMAIL_SUBJECT_PREFIX=[NextGIS QMS Dev feedback]
      - NGID_CLIENT_ID=jQBKtVVaEWTIZT9yZQt9pet6168iTD5sR27dYXT8
      - NGID_CLIENT_SECRET=AlWUFBZXWT0a7wzuFt1rvajgVDuyQXwIOvEWPRVjAJm1Wcx9b0vPTYCPgPfaaqpEGGmZNK9MKNQpfl9n1V2VUwaAG8XrFAybxpCDz1r8m5wvfnQZZdxBZ3aHhmZkVw77
      - ALLOWED_HOST=*
      - OAUTHLIB_INSECURE_TRANSPORT=1
      - IS_DEBUG=false
    command: "./start_socket.sh"
    # command: "./start_http.sh"
#    network_mode: host
    volumes: 
      - socket:/opt/socket
      - django-media:/opt/app/qms_server/media
      - django-static:/opt/app/qms_server/static
      - tmp-nginx:/tmp/nginx
      - nginx-conf:/opt/app/nginx

  nginx:
    depends_on: 
      - qqms
    image: nginx
    user: nginx
#    network_mode: host
    volumes:
      - socket:/opt/socket
      - django-media:/django-media
      - django-static:/django-static
      - tmp-nginx:/tmp
      - nginx-conf:/etc/nginx-conf
      # - /home/sk/py3/quickmapservices_server/docker/custom_files/nginx:/etc/nginx-conf
    ports:
    - "80:8000"
    command: [nginx, '-c', '/etc/nginx-conf/nginx_qms.conf', '-g', 'daemon off;']
volumes:
  db_dir: {}
  socket:
  django-media:
  django-static:
  nginx-conf:
  tmp-nginx:
